app/components/base/advanced-filter/list-filter-control.ts

```diff
@@ -0,0 +1,141 @@
+     * Range need to be speparated by the rangeSeparator
+     */
+    allowRanges?: boolean;
+
+    /**
+     * @default "-"
+     */
+    rangeSeparator?: string;
+}
+
+export class ListFilterControl extends AdvancedFilterControlBase {
+    private _valueControl = new FormControl("");
+
+    constructor(label: string, private _config: ListFilterControlConfig = {}) {
+        super(label);
+        if (!this._config.rangeSeparator) {
+            this._config.rangeSeparator = "|";
+        }
+    }
+
+    public formGroup(): FormGroup {
+        return new FormGroup({
+            type: new FormControl(ListFilterType.Include),
+            value: this._valueControl,
+        });
+    }
+
+    public buildFilter(data: { value: string, type: ListFilterType }): Filter {
+        if (!data.value) {
+            return none();
+        }
+        const values = this.parseValue(data.value);
+        if (data.type === ListFilterType.Include) {
+            const ranges = values.ranges.map(x => {
+                return and(prop(this.name).ge(x[0]), prop(this.name).le(x[1]));
+            });
+            return or(...ranges, prop(this.name).inList(values.items));
+        } else {
+            const ranges = values.ranges.map(x => {
+                return or(prop(this.name).lt(x[0]), prop(this.name).gt(x[1]));
+            });
+            return and(...ranges, prop(this.name).notInList(values.items));
+        }
+    }
+
+    public parseValue(value: string): { items: any[], ranges: any[][] } {
+        if (!value) {
+            return { items: [], ranges: [] };
+        }
+        let values = value.replace(/\s/g, "").split(",").filter(x => x !== "");
+
+        let {items, ranges} = this._extractRanges(values) as any;
+
+        if (this._config.number) {
+            items = this._parseNumbers(items);
+            ranges = this._parseRangeNumbers(ranges);
+        }
+        return {
+            items,
+            ranges,
+        };
+    }
+
+    private _parseNumbers(list: string[]): number[] {
+        return list.map(x => Number(x)).filter(x => !isNaN(x));
+    }
+
+    private _parseRangeNumbers(ranges: string[][]): number[][] {
+        return ranges.map((range) => {
+            return this._parseNumbers(range);
+        }).filter(x => x.length === 2);
+    }
+
+    private _extractRanges(values: string[]) {
+        if (!this._config.allowRanges) {
+            return { items: values, ranges: [] };
+        }
+        const items: string[] = [];
+        const ranges: string[][] = [];
+        for (let v of values) {
+            const range = v.split(this._config.rangeSeparator, 2).filter(x => x !== "");
+            if (range.length === 1) {
+                items.push(v);
+            } else {
+                ranges.push(range);
+            }
+        }
+
+        return {
+            items,
+            ranges,
+        };
+    }
+}
+
+@Component({
+    selector: "bex-adv-filter-list",
+    template: `
```

Comment:
Is it worth putting this in it's own file? Do we have any guidelines on when to use a new file and when to have it inline w/ the component?